velox/expression/tests/TryExprTest.cpp

```diff
@@ -396,5 +401,15 @@ TEST_F(TryExprTest, branchingSpecialForm) {
       "try(switch(always_throws(c6), is_finite(c1), is_finite(c7)))", data);
   expected = makeNullableFlatVector<bool>({true, true, std::nullopt});
   assertEqualVectors(expected, result);
+
+  // Test Switch where the first case produces a partial result while the second
+  // case condition throws at all remaining rows. SwitchExpr may still evaluate
```

Comment:
are we planning to add a change to switch that avoid running the "then" clause for this situation?

Reply:
Added a check to skip the then clause when there is no remaining rows. Thanks!